src/pghstore/_native.py

```diff
@@ -142,9 +141,7 @@ def dump(obj, file, key_map=None, value_map=None, encoding='utf-8'):
     :param encoding: a string encode to use
 
     """
-    if callable(getattr(obj, 'iteritems', None)):
-        items = obj.iteritems()
-    elif callable(getattr(obj, 'items', None)):
+    if callable(getattr(obj, 'items', None)):
```

Comment:
```
~ ❯❯❯ /usr/local/bin/python2.7 -m timeit -n 1000 -s 'd=dict(enumerate(range(1000000)))' 'for i in d.items(): pass'
1000 loops, best of 3: 182 msec per loop
~ ❯❯❯ /usr/local/bin/python2.7 -m timeit -n 1000 -s 'd=dict(enumerate(range(5000)))' 'for i in d.items(): pass'
1000 loops, best of 3: 528 usec per loop
~ ❯❯❯ /usr/local/bin/python2.7 -m timeit -n 1000 -s 'd=dict(enumerate(range(1000000)))' 'for i in d.iteritems(): pass'
1000 loops, best of 3: 82.2 msec per loop
~ ❯❯❯ /usr/local/bin/python2.7 -m timeit -n 1000 -s 'd=dict(enumerate(range(5000)))' 'for i in d.iteritems(): pass'
1000 loops, best of 3: 372 usec per loop
```

Then in a virtualenv with python3.6 and six

```
(venv) ~ ❯❯❯ python3.6 -m timeit -n 1000 -s 'd=dict(enumerate(range(5000)))' 'for i in d.items(): pass'
1000 loops, best of 3: 118 usec per loop
(venv) ~ ❯❯❯ python3.6 -m timeit -n 1000 -s 'import six; d=dict(enumerate(range(5000)))' 'for i in six.iteritems(d): pass'
1000 loops, best of 3: 122 usec per loop
(venv) ~ ❯❯❯ python3.6 -m timeit -n 1000 -s 'd=dict(enumerate(range(1000000)))' 'for i in d.items(): pass'
1000 loops, best of 3: 25.5 msec per loop
(venv) ~ ❯❯❯ python3.6 -m timeit -n 1000 -s 'import six; d=dict(enumerate(range(1000000)))' 'for i in six.iteritems(d): pass'
1000 loops, best of 3: 27.3 msec per loop
```

Timeit setup and usage taken shamelessly from http://lists.openstack.org/pipermail/openstack-dev/2015-June/066391.html although I re-ran all of the above locally.

---

In other words, unless we're using **very** large dictionaries, we won't incur a performance penalty by just using `items()`. As far as I know, we discourage users from going over 100 columns in the first place, so even if someone used 5000 we're talking about 0.0002 second difference in time on Python 2.7 and about 1/50th of that on Python 3.6